test/integration/integration_test.cc

```diff
@@ -24,21 +28,21 @@ TEST_F(IntegrationTest, Echo) {
   EXPECT_EQ("hello", response);
 }
 
-TEST_F(IntegrationTest, RouterNotFound) { testRouterNotFound(Http::CodecClient::Type::HTTP1); }
+TEST_P(IntegrationTest, RouterNotFound) { testRouterNotFound(Http::CodecClient::Type::HTTP1); }
```

Comment:
I followed up with @jamessynge and I believe we settled on explicit as the default for "principle of least surprise".

I'm not sure if this answers your question. For the unparameterized tests, the echo listener is instantiated once with an IPv4 address and once with an IPv6 address. However, the tests are basically just running twice with IPv4 addresses because they are hard coded. My next PR is to plumb through all the integration tests and TODOs in this PR.